scintellometry/io/arochime.py

```diff
@@ -167,6 +168,85 @@ def open(self, number=0):
         return self.fh_raw
 
 
+class AROCHIMEInvPFB(SequentialFile):
+
+    telescope = 'arochime-invpfb'
+
+    def __init__(self, raw_files, blocksize, samplerate, fedge, fedge_at_top,
+                 time_offset=0.0*u.s, dtype='4bit,4bit', comm=None):
+        """ARO data acquired with a CHIME correlator, PFB inverted.
+        """
+        self.fh_raw = AROCHIMERawData(raw_files, blocksize, samplerate,
+                                      fedge, fedge_at_top, time_offset,
+                                      dtype='cu4bit,cu4bit', comm=comm)
+        self.time0 = self.fh_raw.time0
+        self.samplerate = self.fh_raw.samplerate
+        self.dtsample = (1 / self.samplerate).to(u.s)
+        self.npol = self.fh_raw.npol
+        self.fedge = self.fh_raw.fedge
+        self.fedge_at_top = self.fh_raw.fedge_at_top
+        super(AROCHIMEInvPFB, self).__init__(raw_files, blocksize, dtype,
+                                             nchan=1, comm=comm)
+        # PFB information
+        self.nblock = 2048
+        self.h = pfb.sinc_hamming(4, self.nblock).reshape(4, -1)
+        self.fh = None
+        # S/N for use in the Wiener Filter
+        prec = (1 << 3) ** 0.5
+        self.sn = prec / 0.2887
+        print("Opened InvPFB reader, S/N={0}".format(self.sn))
+
+    def open(self, number=0):
+        self.fh_raw.open(number)
+
+    def close(self):
+        self.fh_raw.close()
+
+    def _seek(self, offset):
```

Comment:
The more I thought about this, the less I liked it... We're not going to where we said we would, and this causes problems below with `raw_start`, `raw_end`. I think the easiest solution is to keep `self.recordsize` consistent with the raw one. This implies that although we have a full stream of samples, that the folding code should think of it as sets of 2048 samples. I *think* that all that would be needed to get this working is to change the dtype to an equivalent of `2048u4bit`...